Maker-Agents-Suit-Module/ESMAgent/src/ESM.join.event.agent.ts

```diff
@@ -36,8 +48,25 @@ const provideESMJoinEventAgent = (
   _contractAddress: string,
 ): HandleTransaction => {
   return async (txEvent: TransactionEvent): Promise<Finding[]> => {
+    let findingGenerator: FindingGenerator;
+
+    const logs = txEvent.filterEvent(
+      MAKER_ESM_JOIN_EVENT_SIGNATURE,
+      _contractAddress,
+    );
+
+    if (logs.length) {
+      findingGenerator = createFindingGenerator(
+        _alertID,
+        logs[0].topics[1],
+        BigInt(logs[0].data).toString(),
+      );
+    } else {
+      findingGenerator = createFindingGenerator(_alertID, '', '');
+    }
```

Comment:
This works only if the transaction has only one Join event log. I am going to improve the `provideEventCheckerHandler` to pass the data to the finding generator.